pkg/templates/kubeadm/v1beta2/kubeadm.go

```diff
@@ -272,7 +273,8 @@ func NewConfigWorker(s *state.State, host kubeoneapi.HostConfig) ([]runtime.Obje
 	}
 
 	nodeRegistration := kubeadmv1beta2.NodeRegistrationOptions{
-		Name: host.Hostname,
+		Name:      host.Hostname,
+		CRISocket: s.ContainerRuntimeConfig().CRISocket(),
```

Comment:
Why we don't have Taints here as we do in the control plane config?

Reply:
this is worker config

Reply:
I know, but the behavior should be:

* If it's a control plane node, default taints to the control plane taint
* If it's a worker node, default taints to no taints

However, in both cases, users should be able to provide custom taints (using the `host.Taints` field).

Reply:
This is not directly related to this PR, but if it's a bug, we should fix it or create a follow-up ticket.